docs/guides/player-workflows.md

```diff
@@ -77,6 +89,275 @@ modal.on('hide', function() {
+
+```js
+  var myPlayer = videojs('some-player-id');
+  myPlayer.src("http://www.example.com/path/to/video.mp4");
+  myPlayer.ready(function() {
+     myPlayer.currentTime(10);
+
+     // should be 10 seconds less than duration
+     console.log(myPlayer.remainingTime());
+  });
+```
+
+`buffered` will give you a timeRange object representing the current ranges of time that are ready to be played at a future time.
+
+```js
+  var myPlayer = videojs('some-player-id');
+
+  myPlayer.src("http://www.example.com/path/to/video.mp4");
+  myPlayer.ready(function() {
+    var bufferedTimeRange = myPlayer.buffered();
+
+    // number of different ranges of time have been buffered.
+    // Usually 1
+    var numberOfRanges = bufferedTimeRange.length,
+
+    // Time in seconds when the first range starts.
+    // Usually 0
+    var firstRangeStart = bufferedTimeRange.start(0),
+
+    // Time in seconds when the first range ends
+    var firstRangeEnd = bufferedTimeRange.end(0),
+
+    // Length in seconds of the first time range
+    var firstRangeLength = firstRangeEnd - firstRangeStart;
+  });
+```
+
+`bufferedPercent` will give you the the current percentage of the video that is buffered.
+
+```js
+  var myPlayer = videojs('some-player-id');
+
+  myPlayer.src("http://www.example.com/path/to/video.mp4");
+  myPlayer.ready(function() {
+    // example 0.11 aka 11%
+    var howMuchIsDownloaded = myPlayer.bufferedPercent();
+  });
+```
+
+## Dealing with the source or the poster on the player
+
+Passing a source to the player via the API. (this can also be done using options)
+
+```js
+  var myPlayer = videojs('some-player-id');
+
+  myPlayer.src("http://www.example.com/path/to/video.mp4");
+```
+
+**Source Object (or element):** A javascript object containing information
+about the source file. Use this method if you want the player to determine if
+it can support the file using the type information.
+
+```js
+  var myPlayer = videojs('some-player-id');
+
+  myPlayer.src({type: "video/mp4", src: "http://www.example.com/path/to/video.mp4"});
+```
+
+**Array of Source Objects:** To provide multiple versions of the source so
+that it can be played using HTML5 across browsers you can use an array of
+source objects. Video.js will detect which version is supported and load that
+file.
+
+```js
+  var myPlayer = videojs('some-player-id');
+
+  myPlayer.src([
+    {type: "video/mp4", src: "http://www.example.com/path/to/video.mp4"},
+    {type: "video/webm", src: "http://www.example.com/path/to/video.webm"},
+    {type: "video/ogg", src: "http://www.example.com/path/to/video.ogv"}
+  ]);
+```
+
+Changing or setting the poster via the API. (this can also be done with options)
+
+```js
+  var myPlayer = videojs('example_video_1');
+
+  // set
+  myPlayer.poster('http://example.com/myImage.jpg');
+
+  // get
+  console.log(myPlayer.poster());
+  // 'http://example.com/myImage.jpg'
+```
+
+## Accesing the Tech on the player
```

Comment:
This reminds me, I'm thinking of changing this in 6.0. I want just `tech()` to log a warning along the lines of "I sure hope you know what you're doing" unless the object (or anything) is passes as an argument to `tech()`. It will always return `tech` regardless, it will just log the warning if no argument was passed.